include/matx/core/tensor_impl.h

```diff
@@ -894,6 +896,86 @@ MATX_IGNORE_WARNING_POP_GCC
       return data_.ldata_ + GetValC<0, Is...>(cuda::std::make_tuple(indices...));
     }
 
+    // Locates position of an element at given indices, or returns -1 when not
+    // found.
+    template <int L = 0>
+    __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t
+    GetPos(index_t *lvlsz, index_t *lvl, index_t pos) const {
+      static constexpr int LVL = TensorData::Format::LVL;
+      if constexpr (L < LVL) {
+        using ftype = std::tuple_element_t<L, typename TensorData::Format::LVLSPECS>;
+        if constexpr (ftype::lvltype == ::matx::experimental::LvlType::Dense) {
+          // Dense level: pos * size + i.
+          // TODO: see below, use a constexpr GetLvlSize(L) instead?
+          const index_t dpos = pos * lvlsz[L] + lvl[L];
+          if constexpr (L + 1 < LVL) {
+            return GetPos<L + 1>(lvlsz, lvl, dpos);
+          } else {
+            return dpos;
+          }
+        } else if constexpr (ftype::lvltype == ::matx::experimental::LvlType::Singleton) {
+          // Singleton level: pos if crd[pos] == i and next levels match.
+          if (CRDData(L)[pos] == lvl[L]) {
+            if constexpr (L + 1 < LVL) {
+              return GetPos<L + 1>(lvlsz, lvl, pos);
+            } else {
+              return pos;
+            }
+          }
+        } else if constexpr (ftype::lvltype == ::matx::experimental::LvlType::Compressed ||
+                            ftype::lvltype == ::matx::experimental::LvlType::CompressedNonUnique) {
+          // Compressed level: scan for match on i and test next levels.
+          const typename TensorData::crd_type *c = CRDData(L);
+          const typename TensorData::pos_type *p = POSData(L);
+          for (index_t pp = p[pos], hi = p[pos + 1]; pp < hi; pp++) {
+            if (c[pp] == lvl[L]) {
+              if constexpr (L + 1 < LVL) {
+                const index_t cpos = GetPos<L + 1>(lvlsz, lvl, pp);
+                if constexpr (ftype::lvltype == ::matx::experimental::LvlType::Compressed) {
+                  return cpos; // always end scan (unique)
+                } else if (cpos != -1) {
+                  return cpos; // only end scan on success (non-unique)
+                }
+              } else {
+                return pp;
+              }
+            }
+          }
+        }
+      }
+      return -1; // not found
+    }    
+
+    template <typename... Is>
+    __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ T GetSparseValue(Is... indices) const noexcept { 
+      // Element getter (viz. "lhs = Acoo(0,0);"). Note that due to the compact
```

Comment:
move this at the top of the GetSparseValue() def, rather than inside as implementation comment?